statement.go

```diff
@@ -7,9 +7,15 @@ import (
 	"database/sql/driver"
 )
 
+// SnowflakeStmt represents the prepared statement in driver.
```

Comment:
don't we want to name it `PreparedSnowflakeStmt` then? it looks really similar to `snowflakeStmt` struct defined below

Reply:
No, and there are two reasons:
1. This is a common pattern in go - you have a private struct (name starting with small letter) and public interface (name starting with capital letter), which differentiate only by this first letter. This is something like public interface and private implementation.
2. Go SQL uses name `stmt` in code and `prepared statement` only in docs. We have this convention to enrich with our interfaces by prepending with `Snowflake` (see `driver.Rows` -> `SnowflakeRows` or `driver.Result` -> `SnowflakeResult`).